app/models/miq_request_workflow.rb

```diff
@@ -1042,6 +1042,9 @@ def allowed_storages(_options = {})
     storages = hosts.each_with_object({}) do |host, hash|
       host.writable_storages.each { |s| hash[s.id] = s }
     end.values
+
+    storages.reject!(&:maintenance)
```

Comment:
The comment from @Fryguy got lost in the outdated diff.

> Typically, we have a method on the model at the class level like .available_storages? that would contain this kind of logic. cc @gmcculloug Thoughts?

You mean like the `writable_storages` method referenced a few lines up.  Actually, should this logic be part of the `writable_storages` method so it only returns hosts that are currently writable?  (not read-only, not in maintenance mode)

If so this would mean the the automate best-fit logic would already take this into account since it was changed to use the `writable_storages` method.  Otherwise we need to modify the automate best-fit logic to perform this same check.